### ISSUE TITLE
Accepting an invite where the other party has already left

### ISSUE THREAD

**Issue by germain-gg** [vector-im/element-meta#1000]
#### What would you like to do?

Improving the invite screen when the other party has already left the room when the recipient accepts

#### Why would you like to do it?

It is currently a bit off putting, clicking "accept invite" gives a modal saying "The person who invited you has already left." and does not dismiss the invite

#### How would you like to achieve it?

To automatically reject the invite if we realise we won't be able to join that room ever

**Comment by EECvision** [comment]
Hi @gsouquet, can you assign this issue to me?

**Comment by t3chguy** [comment]
> To automatically reject the invite if we realise we won't be able to join that room ever

We don't get state updates for rooms we are invited to, so this sounds like a spec/server issue rather than a client one?

**Comment by Elliria** [comment]
I've also got a stuck invitation from someone who's not online. If it helps any, I'm a Gitter user who just got switched over to Matrix yesterday. When I arrived here, that chat invitation was there and I got the same message and it's still there today.

(rageshake logs submitted)

![StuckChat](https://user-images.githubusercontent.com/25792925/218907329-416b7642-8f5d-4c85-ba2b-e7711b3e9301.png)

**Comment by josephj11** [comment]
The reason @Elliria is stuck is another small deficiency in the new system. I had two conversations with her - the old one that could not be continued and the automatically generated new one that I could join to continue, but I couldn't tell which was which and deleted the new one by accident. These need to be color coded or made to stand out in some other way so it's easy to tell which is which.

**Comment by waldyrious** [comment]
I'm not sure if this is exactly the same issue, but it kinda matches the description.

In my case, it was a conversation I had created between my gitter user and my Matrix user, and I can't seem to join it now that Gitter has migrated, even though I can join other conversations in similar states. Screen recording:
![Gitter failure to join chat with my Matrix account](https://user-images.githubusercontent.com/478237/218982542-45eab16d-f8f1-482a-a5af-5ae14c6b1016.gif)

For the record, I can still access the conversation from the Matrix/Element.io side — although (possibly related?) it started glitching side a few days ago, in that I get a marker of unread messages but when I click on it I get an error message of not being able to find that message. Screen recording:
![Element.io error in conversation with my Gitter account](https://user-images.githubusercontent.com/478237/218984691-b4d3bd67-3fb1-4e73-ae62-030c2d101b8c.gif)

Let me know if I should open a separate issue!

**Comment by Elliria** [comment]
Okay, I got rid of it. Problem solved for me:

1. See the room with the same name as the invitation:
![JoeInvitationAndRoom](https://user-images.githubusercontent.com/25792925/219134079-984f4aa1-65b8-44be-b8f4-0f1b1c0a61d8.png)
2. Click the three little dots on the right side of the room that you were invited into in the left sidebar:
![LeaveTheRoom](https://user-images.githubusercontent.com/25792925/219134138-4621f38e-435c-4235-9bd7-5a01b0387ea3.png)
3. Click **Leave** in the menu.
4. Click the **Leave** button when you get the warning about not being able to rejoin without an invitation:
![GetTheWarning](https://user-images.githubusercontent.com/25792925/219134173-fe2fd115-f758-4d13-9c4e-a46a625611e8.png)
5. Click the invitation (that didn't get removed yet) in the left panel.
6. Click the **Start chatting** button in the right panel.
7. Click the **OK** button on the "Failed to join" warning in the right panel.
8. Click **Reject** in the right panel.
9. And it's gone, but note that it's in a new **Historical** section at the bottom of the left panel in the event that you wanted to go back to it:
![Gone](https://user-images.githubusercontent.com/25792925/219134855-60f4637d-8065-412f-a98d-14450f60633a.png)

**Comment by Elliria** [comment]
I take it back. It's back. Now it's not showing as a notification, but I'm invited to the room when I sign on and the room is as broken as before. If I leave it, it's still there.

**Comment by t3chguy** [comment]
Closing in favour of https://github.com/vector-im/element-meta/issues/1010 which has a linked Synapse issue

**Comment by Elliria** [comment]
I thought I'd add the comment that it finally went away and hasn't returned, for no known reason. I hadn't been trying anything. It's just finally gone.

**Comment by richvdh** [comment]
> Closing in favour of https://github.com/vector-im/element-meta/issues/1010 which has a linked Synapse issue

#1010 is about rooms that have been *deleted* on the server by a server admin. This sounds like a separate issue to me.

**Comment by Elliria** [comment]
By the way, the room is back, but the little number next to it is gone. It still has the same issue when I try to join the conversation. Interestingly, I tried accessing that room using the **Nheko** desktop client and was able to go into the room and add to its conversation from there, but I still can't join the conversation in the **Element** browser client.

**Comment by richvdh** [comment]
@Elliria this issue is about the message you get when you try to accept an invitation but the other user has already left. I think it's unrelated to whatever you're seeing and I suggest you open a new issue.

**Comment by Elliria** [comment]
No, it started as an invitation and the person who initiated the invitation wasn't online when I got and tried to accept the invitation, which resulted in all of the above.

**Comment by Elliria** [comment]
Also, I believe I had started an issue about it and a moderator suggested I join this one. I could be wrong. I've got quite a few issues going at the moment.

**Comment by JeanPaulLucien** [comment]
Your popup "has already left" is wrong: https://github.com/vector-im/element-web/issues/24161
